src/Components/Components/src/RenderTree/ArrayBuilder.cs

```diff
@@ -97,6 +97,17 @@ internal int Append(T[] source, int startIndex, int length)
         public void Overwrite(int index, in T value)
             => _items[index] = value;
 
+        /// <summary>
+        /// Removes the item at the specified index.
+        /// </summary>
+        /// <param name="index">The index of the item to remove.</param>
+        public void RemoveAt(int index)
+        {
+            Array.Copy(_items, index + 1, _items, index, _itemsInUse - 1 - index);
```

Comment:
Will `index + 1` throw if it's attempting to remove the last element in the array?

Reply:
No, because it will be 0-length.